Mathlib/Logic/Basic.lean

```diff
@@ -160,9 +161,150 @@ end eq_or_ne
 
 theorem by_contradiction {p} : (¬p → False) → p := Decidable.by_contradiction
 
+-- alias by_contradiction ← by_contra
+theorem by_contra {p} : (¬p → False) → p := Decidable.by_contradiction
+
+/-
+In most of mathlib, we use the law of excluded middle (LEM) and the axiom of choice (AC) freely.
+The `decidable` namespace contains versions of lemmas from the root namespace that explicitly
+attempt to avoid the axiom of choice, usually by adding decidability assumptions on the inputs.
+
+You can check if a lemma uses the axiom of choice by using `#print axioms foo` and seeing if
+`classical.choice` appears in the list.
+-/
+--library_note "decidable namespace"
```

Comment:
This shouldn't be too hard to implement either (at least figuring out the syntax).